pkgs/development/python-modules/tree-sitter-grammars/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, buildPythonPackage
+, pytestCheckHook
+, tree-sitter
+, symlinkJoin
+, writeTextDir
+  # `name`: grammar derivation pname in the format of `tree-sitter-<lang>`
+, name
+, grammarDrv
+}:
+let
+  inherit (grammarDrv) version;
+
+  snakeCaseName = lib.replaceStrings [ "-" ] [ "_" ] name;
+  drvPrefix = "python-${name}";
+  langIdentOverrides = {
+    tree_sitter_org_nvim = "tree_sitter_org";
+  };
+  langIdent = langIdentOverrides.${snakeCaseName} or snakeCaseName;
+in
+assert (lib.assertMsg (!(grammarDrv.meta ? license)) ''
+  As of this writing, ${grammarDrv.pname} surprisingly doesn't have a license.
+  This trap is set here to guarantee that if it ever does have a license, this package will inherit the license.
+'');
+buildPythonPackage {
+  inherit version;
+  pname = drvPrefix;
+
+  src = symlinkJoin {
+    name = "${drvPrefix}-source";
+    paths = [
+      (writeTextDir "${snakeCaseName}/__init__.py"
```

Comment:
To be entirely frank, this seems like madness. It's hard to maintain code, just inline in some file, and I'm fairly certain you could have accomplished the same result using https://github.com/grantjenks/py-tree-sitter-languages (which is already packaged) and by putting less than 15-mins of work into https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/tree-sitter/default.nix so it has a package option for additional grammars.

Reply:
[grantjenks/py-tree-sitter-languages](https://github.com/grantjenks/py-tree-sitter-languages) is using a deprecated API:

- code: https://github.com/grantjenks/py-tree-sitter-languages/blob/42f4baffec92848be4937b0cc52b2872201fe322/tree_sitter_languages/core.pyx#L14
- deprecation notice: https://github.com/tree-sitter/py-tree-sitter/tree/v0.21.3#build-from-source
- relevant discussion: https://discourse.nixos.org/t/need-help-enabling-grammars-in-treesitter-python/39500/7